cmd/sync.go

```diff
@@ -104,6 +108,28 @@ func (raw *rawSyncCmdArgs) parsePatterns(pattern string) (cookedPatterns []strin
 	return
 }
 
+func (raw *rawSyncCmdArgs) parseMaxObjectIndexerMapInGb() (uint32, error) {
+
+	// Default case where user not specified any value.
+	if raw.maxObjectIndexerMapSizeInGB == "" {
+		// TODO: Need to add code check system memory and use 30% of that memory.
+		// As of now returning 2 GB.
+		return 2, nil
+	}
+
+	value, err := strconv.Atoi(raw.maxObjectIndexerMapSizeInGB)
+	if value < 0 {
+		err = fmt.Errorf("maxObjectIndexerMapSizeInGb is negative")
+		return 0, err
+	}
+
+	if err != nil {
+		return uint32(value), fmt.Errorf("Parsing failed for maxObjectIndexerMapSizeInGb with error: %v", err)
+	}
```

Comment:
strconv.Atoi return 0 in case of error. So it will not effect that check . But i agree it made code more readable.

Reply:
We need to write this code for windows too, its basically calling system call to get /proc/meminfo in case of linux. We need to handle the case of linux version less than 3.14 where Available Memory not available. Similarily need to do for windows. 
So i am parking it as of now as TODO:.